cl_dll/tri.cpp

```diff
@@ -105,9 +107,47 @@ void DrawAACuboid(triangleapi_s *pTriAPI, Vector corner1, Vector corner2)
 	pTriAPI->End();
 }
 
+void DrawPolyOrCuboid(model_t *model, Vector corner1, Vector corner2)
+{
+	auto pTriAPI = gEngfuncs.pTriAPI;
+
+	if (model && !model->nummodelsurfaces)
+	{
+		DrawAACuboid(pTriAPI, corner1, corner2);
+		return;
+	}
+
+	#ifndef SOFTWARE_BUILD
+	if (IEngineStudio.IsHardware() && model && model->nummodelsurfaces)
+	{
+		#define DrawPolygons(surfs) \
```

Comment:
That's cursed. Put it in a lambda with `auto` parameter (C++14, which OpenAG builds with) or a template function